servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -537,13 +538,13 @@ private TempBlockMeta createBlockMetaInternal(long userId, long blockId,
       if (newBlock) {
         checkTempBlockIdAvailable(blockId);
       }
-      TempBlockMeta tempBlock =
-          mAllocator.allocateBlockWithView(userId, blockId, initialBlockSize, location,
-              getUpdatedView());
-      if (tempBlock == null) {
+      StorageDirView dirView =
+          mAllocator.allocateBlockWithView(initialBlockSize, location, getUpdatedView());
+      if (dirView == null) {
         // Allocator fails to find a proper place for this new block.
         return null;
       }
+      TempBlockMeta tempBlock = dirView.createTempBlockMeta(userId, blockId, initialBlockSize);
```

Comment:
Can you leave a todo here so that we could remove `StorageDirView.createTempBlockMeta` which makes `StorageDirView` completely immutable.